pkg/v11/resource/chartoperator/create.go

```diff
@@ -134,3 +198,25 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 
 	return createState, nil
 }
+
+// checkDeploymentReady checks for the specified deployment that the number of
+// ready replicas matches the desired state.
+func (r *Resource) checkDeploymentReady(ctx context.Context, k8sClient kubernetes.Interface, namespace, deploymentName string, replicas int) error {
+	desc := fmt.Sprintf("'%s'.'%s'", namespace, deploymentName)
+	deploy, err := k8sClient.Extensions().Deployments(namespace).Get(deploymentName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment %s not found", desc))
+		return microerror.Mask(err)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("want %d replicas: %d ready", *deploy.Spec.Replicas, deploy.Status.ReadyReplicas))
+
+	if deploy.Status.ReadyReplicas != *deploy.Spec.Replicas {
+		return microerror.Maskf(notReadyError, fmt.Sprintf("%s has insufficient ready pods", desc))
```

Comment:
It is called `Maskf` with `f` because you can provide a format string and arguments. You do not need to use `fmt.Sprintf`.

Reply:
Ack. I just took this from k8s-migrator I didn't spend lots of time. I'll fix.